resources/js/Pages/Lessons/Show.vue

```diff
@@ -0,0 +1,101 @@
+<script setup>
+import {onMounted, ref} from "vue";
+import unityGameSetUp from '../../GameViewJava';
+import {Head, Link} from '@inertiajs/inertia-vue3';
+
+onMounted(() => {
+    unityGameSetUp();
+    window.addEventListener("stateFromUnity",(e) => recievedJsonFromUnity(e));
+    window.addEventListener("requestData", (e) => requestLoadGameData(e));
+    window.addEventListener("StartChallenge", (e) => recievedChallengeFromUnity(e));
+})
+const emit = defineEmits(['showChallenge'])
+const props = defineProps({
+    lesson: Object ,
+    course: Object,
+
+})
+let showChallenge = ref(false);
+let challengeReponse = ref("hello");
+let challengeAnswer = ref(" ");
+const recievedChallengeFromUnity =  async (e)=>
+{
+    challengeAnswer.value = e.detail.challenge;
+    console.log(challengeAnswer.value);
+    showChallenge.value = true;
+}
```

Comment:
```suggestion
const recievedChallengeFromUnity = async (e) => {
    challengeAnswer.value = e.detail.challenge;
    console.log(challengeAnswer.value);
    showChallenge.value = true;
}
```

The `{` should be inline right after the `async (e) =>`, also be careful with extra spaces. Specially with Javascript as this might break the code at times